src/python/pants/backend/jvm/tasks/jvm_compile/zinc/zinc_compile.py

```diff
@@ -628,7 +629,11 @@ def process_info_file(cp_elem, info_file):
 class ZincCompile(BaseZincCompile):
   """Compile Scala and Java code to classfiles using Zinc."""
 
-  compiler_name = 'zinc'
+  deprecated_module(
+    removal_version='1.19.0.dev0',
+    hint_message="compile.zinc task is being phased out. Use compile.rsc:zinc-only option instead."
```

Comment:
all different scopes. this would be issued out anytime `ZincCompile` module is imported/used as opposed to the other two scopes - `deprecated_options_scope` and `deprecated_conditional` where they will be issued for options usage such as `zinc_compile.xx` and incompatible compiler usage such as `zinc` instead of `rsc` in subsystems respectively.
A quick look at the `hint_message` for each, should clarify more.